lib/autobuild/environment.rb

```diff
@@ -426,15 +426,25 @@ def self.update_environment(newprefix, includes = nil)
             end
         end
 
-        if !includes || includes.include?('LD_LIBRARY_PATH')
+        if macos?
```

Comment:
Since the handling of both variables is basically identical, I would prefer

varname =
  if macos? then 'DYLD_LIBRARY_PATH'
  else 'LD_LIBRARY_PATH'
  end

and then factor out the common code.